test/remote.go

```diff
@@ -48,17 +49,35 @@ func CreateImage(ref string, objs ...runtime.Object) (string, error) {
 			return "", fmt.Errorf("error serializing object: %w", err)
 		}
 
-		layer, err := tarball.LayerFromReader(bytes.NewReader(data))
+		// Compress the data into a tarball.
+		var tarbundle bytes.Buffer
+		writer := tar.NewWriter(&tarbundle)
+		if err := writer.WriteHeader(&tar.Header{
+			Name:     getObjectName(obj),
+			Mode:     0600,
+			Size:     int64(len(data)),
+			Typeflag: tar.TypeReg,
+		}); err != nil {
+			return "", err
+		}
+		if _, err := writer.Write(data); err != nil {
+			return "", err
+		}
+		if err := writer.Close(); err != nil {
+			return "", err
+		}
+
+		layer, err := tarball.LayerFromReader(&tarbundle)
 		if err != nil {
 			return "", fmt.Errorf("unexpected error adding layer to image %w", err)
 		}
 
 		img, err = mutate.Append(img, mutate.Addendum{
 			Layer: layer,
 			Annotations: map[string]string{
-				"org.opencontainers.image.title": getObjectName(obj),
-				"cdf.tekton.image.kind":          strings.ToLower(obj.GetObjectKind().GroupVersionKind().Kind),
-				"cdf.tekton.image.apiVersion":    strings.ToLower(obj.GetObjectKind().GroupVersionKind().Version),
+				"dev.tekton.image.name":       getObjectName(obj),
+				"dev.tekton.image.kind":       strings.ToLower(obj.GetObjectKind().GroupVersionKind().Kind),
+				"dev.tekton.image.apiVersion": strings.ToLower(obj.GetObjectKind().GroupVersionKind().Version),
```

Comment:
Shouldn't those key use the `const` ?

Reply:
This is just my personal opinion, but I prefer to not depend on const values from the unit under test because then you aren't actually testing the value.

Reply:
> This is just my personal opinion, but I prefer to not depend on const values from the unit under test because then you aren't actually testing the value.

We are not testing the value of the `key` — because we define it as a constant, we can rule out that it would change — what we want to test is the the value of the `value` is correct.

Reply:
I actually also view it as a test of the key since the value of the key is a custom annotation name that we define. Therefore, it is important that we are looking for the correct key and receiving the expected value.

Reply:
Actually it's not even in a test. It's not even used as tests (aka we *never* test that the annotation are the correct one), it's used to generate an fake image that can be used in test. So I don't see why not using the const *(but I won't fight too much for using it)*.

Testing the "key" of the annotation is, imo, only useful in the case of a blackbox test (aka one that acts as a consumer that doesn't depend on the contract).

Reply:
I think your last point has convinced me. Unless we blackbox the internals, we can't truly test the contract in this manner and that is what the e2e tests are doing so I will switch to the consts here.